src/main/java/DataBaseInteracting.java

```diff
@@ -0,0 +1,70 @@
+import static org.apache.arrow.vector.types.Types.MinorType.FLOAT4;
+import static org.apache.arrow.vector.types.Types.MinorType.INT;
+import static org.apache.arrow.vector.types.Types.MinorType.VARCHAR;
+
+import java.nio.charset.StandardCharsets;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.arrow.memory.RootAllocator;
+import org.apache.arrow.vector.Float4Vector;
+import org.apache.arrow.vector.IntVector;
+import org.apache.arrow.vector.VarCharVector;
+import org.apache.arrow.vector.VectorSchemaRoot;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.Schema;
+
+
+public class DataBaseInteracting {
+  static final RootAllocator allocator = new RootAllocator(Long.MAX_VALUE);
```

Comment:
You should not have a `static` RootAllocator, because the allocator MUST be closed at the end, by doing this you risk have a memory leak in your program

Reply:
Done.